docs/specification.md

```diff
@@ -1,150 +1,371 @@
 
-* **seqcol digest -> sequence digests**: Given a seqcol digest, sequence digests for all contained sequences can be retrieved by the `/seqcol/:digest/:reclimit` endpoint by setting the `reclimit` to 1.
-* **seqcol digest -> sequences**: Given a seqcol digest, sequences themselves for all contained sequences can be retrieved by the `/seqcol/:digest/:reclimit` endpoint by setting the `reclimit` to 2, or omitting it.
-* **seqcol digest -> metadata**: Retrieved by using the `metadata` endpoint.
-* **seqcol digest -> aliases of seqcol**: Does the API offer this? Should be consistent with refget.
-* **seqcol digest -> metadata of seqcol**: Provided by the `metadata` endpoint.
-* **2 seqcol digests -> assessment of compatibility**: Provided by the compatibility function of the seqcol protocol. First, retrieve the annotated sequence digests for each sequence collection using the *Retrieval* function, then use the *Comparison* function to assess compatibility.
-* **seqcol digest + sequences -> validation claim**: Use the *seqcol algorithm* to compute the digest of the set of sequences, and then use the *comarison* function to validate (or, if you validation requires strict identity, just confirm that the digests match).
 
+### F5. The GA4GH digest algorithm
 
-### Sequence collection as input
+The GA4GH digest algorithm, `sha512t24u`, was created as part of the [Variation Representation Specification standard](https://vrs.ga4gh.org/en/stable/impl-guide/computed_identifiers.html).  This procedure is described as ([Hart _et al_. 2020](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0239883)):
 
-* **sequences -> refget digests**: Individual sequences can be converted into refget digests using the refget algorithm
-* **sequences -> seqcol digest**: Sequence collections can be converted into seqcol digests using the seqcol algorithm. To obtain a sequence digest, follow the algorithm outlined above, sequences -> refget digests -> seqcol digest.
+- performing a SHA-512 digest on a binary blob of data
+- truncate the resulting digest to 24 bytes
+- encodes the 24 bytes using `base64url` ([RFC 4648](https://datatracker.ietf.org/doc/html/rfc4648#section-5)) resulting in a 32 character string
 
-### Refget digest as input
+In Python, the digest can be computed with this function:
 
-* **refget digest -> seqcol digests containing this sequence**: Use the `containing_collections` secondary endpoint. 
-* **many refget digests -> seqcol digests containing all of these sequences**: Repeated application of `containing_collections` endpoint?
+```python
+def trunc512_digest(seq, offset=24):
+    """ GA4GH digest function """
+    digest = hashlib.sha512(seq.encode()).digest()
+    hex_digest = binascii.hexlify(digest[:offset])
+    return hex_digest.decode()
+```
```

Comment:
This routine describes `trunc512` not `sha512t24u`. Should change to.

```suggestion
def sha512t24u_digest(seq):
    """ GA4GH digest function """
    offset = 24
    digest = hashlib.sha512(seq.encode()).digest()
    tdigest_b64us = base64.urlsafe_b64encode(digest[:offset])
    return tdigest_b64us.decode("ascii")
```

This is the same code as is available from https://pypi.org/project/ga4gh.vrs through the routine

```python
import ga4gh.core as c
digest = c.sha512t24u(b"ACGT")
```

Reply:
I didn't realize these were different.